### PR TITLE
dashboard: support token

### LINKED ISSUE #1092: dashboard support authority management
## Feature Request

**Is your feature request related to a problem? Please describe:**

Chaos can be created and deleted by anyone who has access to the dashboard, and the authority of dashboard is very high, it means the Dashboard is a big security hazard.

**Describe the feature you'd like:**

The dashboard should support authority management, login by service account and token created by k8s. And the user can only do the chaos limited by the role bind with the service account.

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve?

RFC: https://github.com/chaos-mesh/rfcs/blob/main/text/2020-10-22-authn-and-authz-on-chaos-dashboard.md

Resolve #1092

Chaos can be created and deleted by anyone who has access to the dashboard, and the authority of the dashboard is very high, which means the Dashboard is a big security hazard.

The dashboard should use the k8s client with a token to visit the k8s cluster, and the token is provided by the user in the dashboard frontend. Users can only do chaos which is allowed by the token generated by the service account.

### What is changed and how does it work?

- Add a k8s client pool implement by LRU, and it will save several clients in the cache.
- HTTP API will extract the token and get k8s client from the client pool.
- The dashboard will be prompt you to add a token when first open.
- Support token switching in the dashboard.
- Support namespace switching in the dashboard.

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Breaking backward compatibility

Related changes

- [x] Need to update the documentation

### Does this PR introduce a user-facing change?

image

image

### CODE REVIEW

**Review by STRRL — CHANGES_REQUESTED** [review]
rest LGTM

**Comment by WangXiangUSTC** [comment]
@STRRL all comment addressed, PTAL again

**Review by STRRL — APPROVED** [review]
LGTM

**Comment by fewdan** [comment]
It looks like you need to deal with the conflict.

**Comment by WangXiangUSTC** [comment]
> It looks like you need to deal with the conflict.

conflict resolved, PTAL again @fewdan

**Review by fewdan — DISMISSED** [review]
LGTM

File: pkg/collector/collector.go [hunk 1]
```diff
@@ -212,8 +212,6 @@ func (r *ChaosCollector) setUnarchivedExperiment(req ctrl.Request, obj v1alpha1.
 		archive.Action = string(chaos.Spec.Action)
 	case *v1alpha1.IoChaos:
 		archive.Action = string(chaos.Spec.Action)
-	case *v1alpha1.DNSChaos:
-		archive.Action = string(chaos.Spec.Action)
```
Comment (STRRL) [inline pkg/collector/collector.go]: Why do we delete this branch for `*v1alpha1.DNSChaos`? I think we should keep it.

Comment (STRRL) [reply pkg/collector/collector.go]: Nevermind.... This is an outdated file.  😰

Comment (WangXiangUSTC) [reply pkg/collector/collector.go]: it's added in another pr https://github.com/chaos-mesh/chaos-mesh/pull/1238

**Review by STRRL — APPROVED** [review]
LGTM

**Review by fewdan — APPROVED** [review]
LGTM

**Pull request merged**